icu4c/source/i18n/dtptngen.cpp

```diff
@@ -1663,6 +1663,9 @@ DateTimePatternGenerator::adjustFieldTypes(const UnicodeString& pattern,
                             && (typeValue!= UDATPG_YEAR_FIELD || reqFieldChar==CAP_Y))
                             ? reqFieldChar
                             : field.charAt(0);
+                    if (c == CAP_E && adjFieldLen < 3) {
```

Comment:
Why is E special here. Is there text in UTS35 mention that? If so, could you add an comment to point to that section in UTS35? Thanks

Reply:
See the lines for "E", "e", and "c" in https://www.unicode.org/reports/tr35/tr35-dates.html#Date_Field_Symbol_Table. 

The description for "e" says "Local day of week number/name, format style. **Same as E except adds a numeric value that will depend on the local starting day of the week.** For this example, Monday is the first day of the week."  The description of the individual groups of letters of different lengths pretty clearly indicate that "c" is intended to work the same as "e".  For "E", on the other hand, "EEE" is the shortest supported length, and "E" is used extensively in the locale data as an alias for "EEE".

I've added a reference to UTS 35 in a comment.

Reply:
Could you put what you wrote above into a comment in the code here? I didn't see "added a reference to UTS 35 in a comment." in this PR yet somehow.